Refactor VOD Domain Module

mostafa-n3ma opened the pull request:
✅ What was done:
•	Merged the previously separated domain modules:
•	actors, movies, series, and search
•	→ into a single unified module: vod_domain
•	Standardized domain structure with:
•	model/ → Core models: Movie, TvSeries, Actor, etc.
•	repository/ → Clean repository interfaces
•	usecase/ → Focused use cases for search and history

🧹 Cleanup:
•	Removed unused domain modules from settings.gradle.kts and build.gradle.kts
•	Renamed domain root folder from Domains → domains for consistency
•	Ensured vod_domain is a pure Kotlin module (no Android dependencies)

🔁 Model Enhancements:
•	Replaced Long timestamp with LocalDateTime using kotlinx.datetime
•	Defined MediaFilters and SearchHistoryItem for scalable search logic
•	Ensured all models are reusable across multiple features (not just search)

💡 Why this change?

This refactor:
•	Centralizes all video-on-demand logic under one cohesive domain
•	Improves maintainability and reusability across features
•	Reduces duplication and confusion from overly granular domain modules

if you have any notes please let me know ....

mostafa-n3ma committed:
delete movie domain module

mostafa-n3ma committed:
delete actor domain module

mostafa-n3ma committed:
delete series domain module

mostafa-n3ma committed:
update project gradle and settings file

mostafa-n3ma committed:
rename search module to vod module and update root gradle and settings files accordingly

mostafa-n3ma committed:
add models

mostafa-n3ma committed:
add repositories

mostafa-n3ma committed:
add use cases

mostafa-n3ma committed:
add toml file support

haideraljawasim commented on domain/vod/src/main/java/model/MediaFilters.kt:
```diff
@@ -0,0 +1,7 @@
+
+data class MediaFilters(
+    val releaseYear: Int? = null,
+    val genres: List<String>? = null,
```
The genres should be enum class , not strings

mostafa-n3ma replied:
All the genres would be like 20 genres I don't think it's right to create enum for that

mostafa-n3ma replied:
sorry it is a good idea 
done ✅

haideraljawasim commented on domain/vod/src/main/java/model/MediaFilters.kt:
```diff
@@ -0,0 +1,7 @@
+data class MediaFilters(
+    val releaseYear: Int? = null,
+    val genres: List<String>? = null,
+    val imdbRating: Float? = null
```
I think we should use Double here instead of Float

mostafa-n3ma replied:
The imdb rating is from 0 to 99.9 it won't be more then that so I think float is enough

mostafa-n3ma replied:
IMDB ratings are limited in range and precision, and we’re not doing complex calculations here, so Float keeps the model lightweight. If we later see any need for more precision, we can refactor

haideraljawasim commented on domain/vod/src/main/java/model/SearchHistoryItem.kt:
```diff
@@ -0,0 +1,11 @@
+    val query: String,
+    val category: SearchCategory,
+    val filters: MediaFilters? = null,
+    val language: String,
```
here also i think it should be enum class for language mapping will be easier and safer

mostafa-n3ma replied:
Agree that would be better 👍🏻

mostafa-n3ma replied:
done ✅

haideraljawasim commented on domain/vod/src/main/java/model/TvSeries.kt:
```diff
@@ -0,0 +1,14 @@
+    val imageUrl: String,
+    val title: String,
+    val description: String,
+    val seasons: List<String>,
```
why would you used here a list of string why didn't you create data class represent season with list episodes for each season or maybe a later get them by series id  which i think is the best approach , cause actually don't get it why list of string

mostafa-n3ma replied:
Agree that would be better nice catch 👍🏻

mostafa-n3ma replied:
done ✅

haideraljawasim commented on domain/vod/src/main/java/model/TvSeries.kt:
```diff
@@ -0,0 +1,14 @@
+    val seasons: List<String>,
+    val actors: List<String>,
+    val releaseDate: LocalDate,
+    val genres: List<String>,
```
same genres should be enum here

mostafa-n3ma replied:
Let me check on that first

haideraljawasim commented on domain/vod/src/main/java/model/TvSeries.kt:
```diff
@@ -0,0 +1,14 @@
+    val title: String,
+    val description: String,
+    val seasons: List<String>,
+    val actors: List<String>,
```
Same why list of strings why not List of Actors ID's or may we get actors later by movie id  since if you entered a movie and then clicked the actors you would not see just it name you will see its image and charter add to the name , so we are literally will not get any thing from this list

haideraljawasim commented on domain/vod/src/main/java/model/movie.kt:
```diff
@@ -0,0 +1,13 @@
+package model
+import kotlinx.datetime.LocalDate
+
+data class Movie(
```
It miss movie duration here

haideraljawasim replied:
also it miss if the movie saved or not the user should be able to save movie , we would need this later also since this a shared entity i think we should add it now

haideraljawasim replied:
also you should add trailer url

mostafa-n3ma replied:
We can save content 🤨!

haideraljawasim replied:
we can save content to watch later you can check the design  i mean not to download but like add in saved list  
<img width="332" alt="Screenshot 2025-07-08 at 1 23 10 PM" src="https://github.com/user-attachments/assets/76395bd5-b2c6-4f8a-9763-177e593d092f" />
from seach screen where the icon on the image represent if the movie or series  saved or not

haideraljawasim commented on domain/vod/src/main/java/model/movie.kt:
```diff
@@ -0,0 +1,13 @@
+
+data class Movie(
+    val id: String,
+    val imageUrl: String,
```
i suggest posterImageUrl the name will be more clear

haideraljawasim commented on domain/vod/src/main/java/model/movie.kt:
```diff
@@ -0,0 +1,13 @@
+    val imageUrl: String,
+    val title: String,
+    val description: String,
+    val actors: List<String>,
```
same i don't get why list of string we are literally would not get any thing useful from this list

mostafa-n3ma replied:
i though of going with list of actors since we are going to use this in detailes screen two but this model would become heavy
mybe list of actors id's is better

haideraljawasim commented on domain/vod/src/main/java/model/SearchCategory.kt:
```diff
@@ -0,0 +1,3 @@
+package model
+
+enum class SearchCategory { MOVIE, TV_SERIES, ACTOR }
```
Not necessary comment but would be more readable if like each value with new line

haideraljawasim commented on domain/vod/src/main/java/model/Actor.kt:
```diff
@@ -0,0 +1,10 @@
+package model
```
we have been told a lot in public code reviews to not call our entities model so i think you should rename this package to entities

mostafa-n3ma replied:
I believe it is the opposite no entities 😅 they are just models

ZainabAlnajjar replied:
@mostafa-n3ma they are entities for the domain layer, 
the mentors made it clear in many code reviews, so please rename it
(even yesterday if you could remember when Bareq said: "no need to add the suffix entity to the data classes like MovieEntity or ActorEntity, but name it directly as Movie or Actor would be enough cuz they are in the entity package")

mostafa-n3ma replied:
Got it rename the packadge to entity @ZainabAlnajjar

haideraljawasim commented on domain/vod/src/main/java/model/Actor.kt:
```diff
@@ -0,0 +1,10 @@
+    val id: String,
+    val imageUrl: String,
+    val name: String,
+    val age: Int?,
+    val region: String?,
+    val lastShow: String?
```
These values we don't need them also Im not sure if we should keep or remove them but i suggest removing them since we may can't get this info from api

AhmadKharfan replied:
@haideraljawasim 
Our domain model should reflect what an Actor means to the product, not what the current API provides.  
The  region and lastShow are nullable for now if future features never use them, we can safely drop them later.

haideraljawasim reviewed (changes requested):
Well done Mustafa you may need small refactoring ,
Last thing i think you should add some exceptions in this PR if we gonna work on interface implementation  or ui today later

AhmadKharfan commented on domain/vod/src/main/java/model/MediaFilters.kt:
```diff
@@ -0,0 +1,7 @@
+package model
+
+data class MediaFilters(
```
MediaFilters looks like a UI-state holder (user’s current filter choices), not a domain object or VOD domain.
Unless a use-case needs it, move it to the presentation layer (UiState) and keep the domain layer for business entities only.

mostafa-n3ma replied:
Media filters will be saved as properties in the searchHistoryItem so we can get the same filtered data or make new api cal when the filters change for the same query

AhmadKharfan replied:
I see but searchHistoryItem itself should live in UI state it only stores UI context, and the domain layer should stay business only

AhmadKharfan commented on domain/vod/src/main/java/model/SearchHistoryItem.kt:
```diff
@@ -0,0 +1,11 @@
+
+import kotlinx.datetime.LocalDateTime
+
+data class SearchHistoryItem(
```
SearchHistoryItem is screen-level data (just what the user typed + filters) and isn’t reused by core VOD business rules, so it belongs in UI state / presentation, not in the vod domain module.

mostafa-n3ma replied:
We need to save history locally in data base 
Also call this in the CRUD useCases

mostafa-n3ma replied:
Maybe we need to discuss it today in the meeting

AhmadKharfan replied:
We can keep SearchHistoryItem as a UI-state object and let each CRUD use-case expose its own Params (or nested data class) the presentation layer does the mapping, then hands the domain an inner data class that the use-case owns

```
// ---------- presentation (ViewModel) ----------
val uiState = SearchHistoryItem(query = "Matrix", filters = genreFilters)

// Map UI state ➜ use-case params
val params = SearchMoviesUseCase.Params(
    query   = uiState.query,
    filters = uiState.filters
)

searchMoviesUseCase(params)


// ---------- domain ----------
class SearchMoviesUseCase(
    private val repo: SearchRepository
) {

    // Inner data class visible only to the use-case
    data class Params(
        val query: String,
        val filters: List<Filter>
    )

    operator fun invoke(params: Params) =
        repo.search(params.query, params.filters)
}
```

So:

SearchHistoryItem stays in the UI state.

The ViewModel converts it to Params (the inner data class).

The domain layer receives only its own primitives and never depends on presentation or data storage.

AhmadKharfan replied:
Let’s talk and discuss more in the daily

AhmadKharfan commented on domain/vod/src/main/java/repository/ActorRepository.kt:
```diff
@@ -0,0 +1,7 @@
+import model.Actor
+
+interface ActorRepository {
+    suspend fun searchActors(query: String, language: String): List<Actor>
```
this searchActors() method belong in a SearchRepository within the search domain, not in the core Actor repository.

mostafa-n3ma replied:
I agree that searchActors() is more about search logic, not actor management itself.
I’ll move searchActors() to a SearchRepository and keep ActorRepository focused on actor-specific actions like addToFavorites() and getActorDetails()

ZainabAlnajjar replied:
@AhmadKharfan just to clarify on what Mostafa said, there is no such thing as search domain or search repository as we discussed yesterday

searching is a use case that can deal with many repos, if you search for actors? by actor repo, search for movie? by movie repo, and so on

mostafa-n3ma replied:
I was wrong @ZainabAlnajjar this is not actor CRUD Logic we are not adding or deleting any actor data we are just searching so in the search repository implementation we can use the actor data source and retrive data 
so search repository is clear

ZainabAlnajjar replied:
he deleted the old reply haha

AhmadKharfan replied:
@ZainabAlnajjar We discussed that search isn’t its own business domain, but it’s fine to keep a repository for the search logic.

AhmadKharfan commented on domain/vod/src/main/java/repository/MovieRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import model.Movie
+
+interface MovieRepository {
+    suspend fun searchMovies(query: String, filters: MediaFilters?, language: String): List<Movie>
```
Same for searchMovies() method it should be in SearchRepository within the search domain.

mostafa-n3ma replied:
Same replay please check my replay on the past comment

AhmadKharfan commented on domain/vod/src/main/java/repository/TvSeriesRepository.kt:
```diff
@@ -0,0 +1,8 @@
+import model.TvSeries
+
+interface TvSeriesRepository {
+    suspend fun searchTvSeries(query: String, filters: MediaFilters?, language: String): List<TvSeries>
```
Also here this method belong to  SearchRepository.

mostafa-n3ma commented:
Thank you guys for the valuable notes I will work on them immediately 👍🏻👍🏻👍🏻

mostafa-n3ma committed:
update: genres property type to list of enum Genre

mostafa-n3ma committed:
update: add helper types classes and enums

mostafa-n3ma committed:
update: reformat packages

mostafa-n3ma committed:
update: rename imageUrl to posterImageUrl

mostafa-n3ma committed:
update:
- rename and actors to actorId
- remove movie series and actors repositories and complained there logic in one SearchRepository
- update useCases accordingly and use searchRepository

mostafa-n3ma commented:
dear @haideraljawasim  and @AhmadKharfan 
Thanks for the valuable review; it helped improve the code 
Please check the latest updates

ElhamFadel commented on domain/vod/src/main/java/repository/SearchHistoryRepository.kt:
```diff
@@ -0,0 +1,9 @@
+import model.SearchHistoryItem
+
+interface SearchHistoryRepository {
+    suspend fun getSearchHistory(): List<SearchHistoryItem>
```
```suggestion
    suspend fun getSearchHistory(): Flow<List<SearchHistoryItem>>
```

AhmadKharfan replied:
The Figma design doesn’t show live-updating search, so a simple `suspend` fetch is enough lighter and avoids the extra overhead of streaming data with `Flow`

mostafa-n3ma replied:
just like Ahmed says 👌🏻

AhmadKharfan reviewed (commented):
Great job, Mostafa your code looks great!
We can talk more about the SearchHistoryItem and filters in the daily.
Aashet eidak :)

devfalah commented on domain/vod/src/main/java/model/Language.kt:
```diff
@@ -0,0 +1,11 @@
+    ENGLISH("en"),
+    ARABIC("ar");
+
+    companion object {
+        fun fromCode(code: String): Language? =
+            Language.entries.find { it.code == code }
+    }
```
There's no need for this if you need to map data to a domain, you can create a mapper

devfalah commented on domain/vod/src/main/java/usecase/SearchActorsUseCase.kt:
```diff
@@ -0,0 +1,29 @@
+                SearchCategory.ACTOR,
+                null,
+                language,
+                Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())
```
Create a reusable function that gets the current date time.

```kotlin
fun LocalDateTime.Companion.now(
    timeZone: TimeZone = TimeZone.currentSystemDefault()
) = Clock.System.now().toLocalDateTime(timeZone)

//Usage:
LocalDateTime.now()
```

devfalah commented on domain/vod/src/main/java/usecase/SearchMoviesUseCase.kt:
```diff
@@ -0,0 +1,30 @@
+                SearchCategory.MOVIE,
+                filters,
+                language,
+                Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault())
```
same

AimanYosofiUlt commented on domain/vod/src/main/java/model/Genre.kt:
```diff
@@ -0,0 +1,15 @@
+package model
+
+enum class Genre {
```
I disagree with putting the Genre as an enum class, Genres can change over time, and there are so many Genres
I can argue we should have the ANIME genre here

ZainabAlnajjar replied:
as Aiman said, enum is not the best way to handle genres, there could be more or less types in the future, in that case we will need to edit on all layers just to add the new genre and we should specify every think in static way

I am suggesting doing something like the default categories in Tudee, like the following:

we will have a genre class that has the (id, name, imgPath), the image would come from asset file and will be mapped in the repository

in this case we can search by the genre id in filters and the localization of the name would come from the api itself
another advantage is that even if we changed the Api later the other layers won't get effected, everything would be handled from the DataSource and mapper without the other layers, another thing is that the genre of the movies has different names and ids than the. so making a data class would be more flexible

haideraljawasim replied:
What do you mean by “we will need to edit all layers”?

Adding a new Genre as a data class will not affect  the reset ? like how so 
also adding new genre to enum class will not effect the whole logic of layers it will just effect the mapping between them like just add one line to the When statement to get the new Genre and i think this completely normal 
If you meant just returning the genre name as a String from the data source  that’s also not correct, because that would make the domain layer depend on the data and break encapsulation
the domain should have its own expected structure and not rely on the internal format of the data.

@ZainabAlnajjar

ZainabAlnajjar replied:
@haideraljawasim the domain shouldn't get changed when a new data came along
the enum class would be useful for a static values like task status, but genres could change over time

also, the api have the localization name so that would be another advantage

haideraljawasim replied:
suppose the api return "blah" but ment to  be hmmm let say Action  Genre and our domain required from us to filter by specified genre 
what to do here in this case ?
do we display the genre as returned from api ?

NOTE the return value from API change by time or data source

haideraljawasim replied:
we defiantly will need to map to something that we can use and display as we want

AhmadKharfan committed:
rename model package in vod domain to entity package

AhmadKharfan committed:
move search history item and media filters to params package inside usecase because it's not an entity

AhmadKharfan committed:
move search history item and media filters to params package inside usecase because it's not an entity

AhmadKharfan committed:
add remove search history item function to history repository then add search history htiem usecase

AhmadKharfan committed:
add recent watching methods in history repository and add usecases for curd operations on recent watching item

AhmadKharfan committed:
create extension function in local date to get the current time and use it in usecases

AhmadKharfan committed:
remove mapper from language enum class in domain models

haideraljawasim commented on domain/vod/src/main/java/usecase/AddRecentWatchingUseCase.kt:
```diff
@@ -0,0 +1,10 @@
+import repository.SearchHistoryRepository
+import usecase.search.RecentWatchingItem
+
+class AddRecentWatchingUseCase(
```
Rename recent watching to recent viewed as the designer make it clear this part for recent viewed

haideraljawasim commented on domain/vod/src/main/java/usecase/RemoveSearchHistoryItemUseCase.kt:
```diff
@@ -0,0 +1,12 @@
+class RemoveSearchHistoryItemUseCase(
+    private val historyRepo: SearchHistoryRepository
+) {
+    suspend fun execute(item: SearchHistoryItem) {
+        historyRepo.removeSearchHistoryItem(item)
+    }
```
I'm not get it here why we delete by object why not by id

mostafa-n3ma replied:
As haider says 
It is already checking item by id under the hood 
So by id is straight forward

AhmadKharfan replied:
Actually i thought about deleting by id
but i don’t see an id inside SearchHistoryItem since we’re passing the whole object.
i’m thinking of updating the data class to include an id, but wanted to discuss it with you first.
anyway i’ll go ahead and add the id to the data class and pass it through the use-case

haideraljawasim commented on domain/vod/src/main/java/usecase/search/MediaFilters.kt:
```diff
@@ -0,0 +1,7 @@
+
+data class MediaFilters(
+    val releaseYear: Int? = null,
+    val genres: List<String>? = null,
```
genres we already switched to use enum class for them but here still list of string why ?

AhmadKharfan replied:
We already switched it in the TvSeries entity, but we forgot to update MediaFilters I’ll fix that 😅

AhmadKharfan committed:
rename recent watching to recent viewed in data class and usecases

AhmadKharfan committed:
replace list of genres in media filters from string to enum be list of enum of genre

AhmadKharfan committed:
add id field to search history item and pass it to delete search history item usecase instead of passing the object

AhmadKharfan committed:
make the remove search history item method take id as parameter

haideraljawasim commented on domain/vod/src/main/java/usecase/search/MediaFilters.kt:
```diff
@@ -0,0 +1,9 @@
+
+import entity.Genre
+
+data class MediaFilters(
+    val releaseYear: Int? = null,
```
I checked the design again now and noticed the year filter should be range and not one release year so we should refactor this

ZainabAlnajjar commented on domain/vod/src/main/java/entity/Actor.kt:
```diff
@@ -0,0 +1,10 @@
+package entity
+
+data class Actor(
+    val id: String,
```
why the id is string? it should be long, same for the other entities (even the API returns number)

AhmadKharfan committed:
make the id in search history item nullable because it will generated in room data base and change the id in other entities from string to be long

AhmadKharfan committed:
change filter by year from certain year to be range of years

AhmadKharfan committed:
add search history output data class that will come from the room database

AhmadKharfan committed:
add search history output data class that will come from the room database

AhmadKharfan committed:
add missing genres to the enum class

AhmadKharfan committed:
add title and is saved flag to search history output item data class

ZainabAlnajjar commented on domain/vod/src/main/java/entity/Actor.kt:
```diff
@@ -0,0 +1,10 @@
+package entity
+
+data class Actor(
```
we also need the gender of actor, maybe add it as isMale boolean

AhmadKharfan committed:
add gender field to actor entity

ZainabAlnajjar commented on domain/vod/src/main/java/entity/Genre.kt:
```diff
@@ -0,0 +1,28 @@
+package entity
+
+enum class Genre {
```
missing Science Fiction

ZainabAlnajjar replied:
also `Action & Adventure`

ZainabAlnajjar replied:
SCI_FI -> Sci-Fi & Fantasy

AhmadKharfan committed:
add missing genres to the enum data class

AhmadKharfan committed:
remove gender default value from actor entity

Commit:
add unit test for the vod use cases

Commit:
Merge remote-tracking branch 'origin/task/create-vod-domain-module' into task/create-vod-domain-module

ZainabAlnajjar commented on domain/vod/src/main/java/usecase/search/SearchHistoryOutputItem.kt:
```diff
@@ -0,0 +1,8 @@
+data class SearchHistoryOutputItem(
+    val id: Long? = null,
+    val title: String,
+    val posterImageUrl: String,
```
actor doesn't have poster image; it would be better to just name it `imageUrl`

ZainabAlnajjar replied:
or it would be better to make another output for the actor cuz it also don't have the isSaved parameter

ZainabAlnajjar commented on domain/vod/src/main/java/usecase/SearchTvSeriesUseCase.kt:
```diff
@@ -0,0 +1,29 @@
+        historyRepo.addSearchHistoryItem(
+            SearchHistoryInputItem(
+                query = query,
+                category = SearchCategory.MOVIE,
```
why SearchCategory is movie in the TvSeries search usecase, please change it

ZainabAlnajjar commented on domain/vod/src/main/java/usecase/SearchActorsUseCase.kt:
```diff
@@ -0,0 +1,28 @@
+        historyRepo.addSearchHistoryItem(
+            SearchHistoryInputItem(
+                query = query,
+                category = SearchCategory.MOVIE,
```
same here change to actor

AhmadKharfan committed:
create search actor history output and use cases to get actor history and one to get movie history

AhmadKharfan committed:
replace the wrong category in search movie and actor use case with correct category

Commit:
add unit test for get actor\movie search history

Commit:
add changes in search movies unit-test

AhmadKharfan committed:
use search history media and actor output in search repository and delete unused usecases

haideraljawasim commented on domain/vod/src/main/java/repository/SearchRepository.kt:
```diff
@@ -0,0 +1,21 @@
+import usecase.search.SearchMediaHistoryOutput
+
+interface SearchRepository {
+    suspend fun searchActors(query: String, language: Language): List<SearchActorHistoryOutput>
+    suspend fun searchMovies(
+        query: String,
+        filters: MediaFilters?,
+        language: Language
+    ): List<SearchMediaHistoryOutput>
```
change the name to SearchActorOutput

Commit:
Merge remote-tracking branch 'origin/task/create-vod-domain-module' into task/create-vod-domain-module

AhmadKharfan committed:
change name in search output for media and actor

Commit:
Merge remote-tracking branch 'origin/task/create-vod-domain-module' into task/create-vod-domain-module

AhmadKharfan committed:
change return type in usecases to be output for media and actor

AhmadKharfan committed:
Merge remote-tracking branch 'origin/task/create-vod-domain-module' into task/create-vod-domain-module

AhmadKharfan committed:
change return type in usecases to be output for media and actor

Commit:
add changes in search movies unit-test

ZainabAlnajjar committed:
refactor getSearchHistory to optimize data retrieve

ZainabAlnajjar committed:
fix wrong type reference in domain entities

ZainabAlnajjar committed:
exclude entity from test coverage

mostafa-n3ma committed:
update : excludes entities and data classes from tests

AimanYosofiUlt commented on domain/vod/src/main/java/entity/Genre.kt:
```diff
@@ -0,0 +1,30 @@
+    ACTION,//
+    ADVENTURE,//
+    COMEDY,//
+    DRAMA,//
```
What is the tow slash for?

ZainabAlnajjar committed:
improve tests functionality and readability

ZainabAlnajjar committed:
Merge remote-tracking branch 'origin/task/create-vod-domain-module' into task/create-vod-domain-module

ZainabAlnajjar committed:
remove redundant code

ZainabAlnajjar committed:
clean code

haideraljawasim commented on domain/vod/src/main/java/entity/Genre.kt:
```diff
@@ -1,30 +1,29 @@
+    SOAP,
+    TALK,
+    WAR,//   WESTERN,
+    TV_MOVIE,
```
I think you commented WESTERN by mistake

ZainabAlnajjar replied:
my bad, fixed that

Ahmed7osny1 commented on domain/vod/src/main/java/extensions/DateTimeExtensions.kt:
```diff
@@ -0,0 +1,10 @@
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+
+fun LocalDateTime.Companion.now(
```
- i suggest to change now to current to avoid confusion with existing LocalDateTime.now()

Ahmed7osny1 commented on domain/vod/src/main/java/repository/SearchRepository.kt:
```diff
@@ -0,0 +1,21 @@
+
+interface SearchRepository {
+    suspend fun searchActors(query: String, language: Language): List<SearchActorOutput>
+    suspend fun searchMovies(
+        query: String,
+        filters: MediaFilters?,
+        language: Language
+    ): List<SearchMediaOutput>
+
+    suspend fun searchTvSeries(
+        query: String,
+        filters: MediaFilters?,
+        language: Language
+    ): List<SearchMediaOutput>
```
i think we have two type (tv and movies) and these two fun the same 
we can provide enum
enum class MediaType {
    MOVIE, TV_SERIES
}
and write one methods but take type from  MediaType

Ahmed7osny1 commented on domain/vod/src/main/java/usecase/SearchActorsUseCase.kt:
```diff
@@ -0,0 +1,28 @@
+
+class SearchActorsUseCase(
+    private val searchRepository: SearchRepository,
+    private val historyRepo: SearchHistoryRepository
```
i prefer to change this to historyRepository for consistancy

Ahmed7osny1 commented on domain/vod/src/main/java/usecase/SearchMoviesUseCase.kt:
```diff
@@ -0,0 +1,30 @@
+
+class SearchMoviesUseCase(
+    private val searchRepository: SearchRepository,
+    private val historyRepo: SearchHistoryRepository
```
i prefer to change this to historyRepository for consistancy

Ahmed7osny1 commented on domain/vod/src/main/java/usecase/SearchTvSeriesUseCase.kt:
```diff
@@ -0,0 +1,30 @@
+
+class SearchTvSeriesUseCase(
+    private val searchRepository: SearchRepository,
+    private val historyRepo: SearchHistoryRepository
```
i prefer to change this to historyRepository for consistancy

Ahmed7osny1 commented on domain/vod/src/main/java/entity/Language.kt:
```diff
@@ -0,0 +1,6 @@
+package entity
+
+enum class Language(val code: String) {
```
we can use language provider

Ahmed7osny1 commented on domain/vod/src/main/java/usecase/search/SearchHistory.kt:
```diff
@@ -0,0 +1,9 @@
+import kotlinx.datetime.LocalDateTime
+
+data class SearchHistory (
+    val id: Long? = null,
```
why nullable

Ahmed7osny1 commented on domain/vod/src/main/java/usecase/search/SearchHistoryInputItem.kt:
```diff
@@ -0,0 +1,13 @@
+import kotlinx.datetime.LocalDateTime
+
+data class SearchHistoryInputItem(
+    val id: Long? = null,
```
same thing

Ahmed7osny1 commented on domain/vod/src/main/java/usecase/search/SearchMediaOutput.kt:
```diff
@@ -0,0 +1,8 @@
+package usecase.search
+
+data class SearchMediaOutput(
+    val id: Long? = null,
```
same thing
